### ISSUE TITLE
Graphene ObjectType Constructor - expected no arguments to constructor error

### ISSUE THREAD

**Issue by roshanjrajan-zip** [microsoft/pyright#5277]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**

Starting from Pyright version 1.1.307, when inheriting graphene.ObjectType, the following error is produced- `test.py:7:5 - error: Expected no arguments to "TestObject" constructor (reportGeneralTypeIssues)`. Before, the code was considered fine by Pyright in version < 1.1.306.

**To Reproduce**
Steps to reproduce the behavior.

Using the code below as test.py, run the following command. It fails for versions > 1.1.307 but not for versions < 1.1.306.
`PYRIGHT_PYTHON_FORCE_VERSION=1.1.307 pyright test.py`

**Expected behavior**

Not certain if this is expected behavior or not. The underlying graphene.ObjectType calls into `dataclass = make_dataclass(name_, fields, bases=())` to generate the correct `__init__` function. We know it works because the example below shows the data being passed in and printed out. I would expect Pyright to understand that the constructor accepts arguments.

**Screenshots or Code**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or imports all referenced symbols.

```python
import graphene

class TestObject(graphene.ObjectType):
    var1 = graphene.Int()
    var2 = graphene.String()

a = TestObject(var1=0, var2="hello") # test.py:7:5 - error: Expected no arguments to "TestObject" constructor (reportGeneralTypeIssues)
print(a.var1, a.var2)
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

Using graphene==3.1.1

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.
Pyright through CLI.

**Additional context**

**Comment by erictraut** [comment]
Starting in 1.1.307, the constructor type evaluation logic was overhauled to more accurately conform to the runtime constructor behavior. Prior to that, it was probably just skipping validation of constructor calls in your code sample above. Pyright is working as intended here, so this isn't a bug.

I'm not familiar with graphene, but it appears that it's doing some tricky runtime "magic" similar to the stdlib dataclass, which synthesizes an `__init__` method based on declared instance variables. That means it would probably work with PEP 681 (`dataclass_transform`). Graphene is unfortunately untyped. If the library is important to you, I recommend reaching out to the maintainers and asking if they're open to adding type annotations.

In the meantime, your other options are:
1. Set `useLibraryCodeForTypes` to false. This will treat all symbols imported from graphene (and other untyped libraries) as `Unknown`. This will eliminate the error you're seeing here, but it will also eliminate all type checking and language server features related to these symbols.
2. Find type stubs for graphene. I'm not sure if they exist.
3. Write your own type stubs for the parts of graphene that you use. You could start with a minimal stub consisting of a module-level `__getattr__` method that returns an `Any` type. You could then add in types for more symbols as you see fit.
4. Suppress specific errors using `# type: ignore` or `# pyright: ignore` comments in your code.

Here's how you'd apply PEP 681 to the `ObjectTypeMeta` class in graphene:
```python
@dataclass_transform()
class ObjectTypeMeta(BaseTypeMeta):
    ...
```

With this in place, you code would then type check with some small changes:

```python
class TestObject(ObjectType):
    var1: Int = Int()
    var2: String = String()

a = TestObject(var1=Int(0), var2=String("hello"))
print(a.var1, a.var2)
```

**Comment by roshanjrajan-zip** [comment]
Thanks for your help!
